### DISCUSSION TITLE
how to create DAG for mssql to google cloud storage

### DISCUSSION THREAD

**Discussion by irfanajr17** [apache/airflow#25276] [Q&A]
Hi all
please help to make a DAG example in airflow, to extract data from sql server to csv file in google cloud storage,
I'm having a failed mssql module in my airflow.
Please help me.

![image](https://user-images.githubusercontent.com/109498315/180740551-90f25504-a38a-40ca-bf68-93ec88d9be04.png)
![image](https://user-images.githubusercontent.com/109498315/180740854-55dc084e-c5bc-4cc4-8ba8-1343b05d9c84.png)
![image](https://user-images.githubusercontent.com/109498315/180740978-479fe1a3-80b2-4898-a338-0e11612c84a1.png)

**Comment by potiuk** [comment]
You should install Mssql provider.https://airflow.apache.org/docs/apache-airflow-providers-microsoft-mssql/stable/index.html

**Reply by irfanajr17** [reply]
mssql provider has been installed but shows the same error

![image](https://user-images.githubusercontent.com/109498315/180916690-71876c4e-a850-455e-ab2d-d101a99799de.png)
![image](https://user-images.githubusercontent.com/109498315/180916733-d26cad09-5270-4727-ad77-5a661b1fffc9.png)

**Reply by potiuk** [reply]
You know that Airflow does not work on Windows and you should use WSL2 / linux filesystem if you want to run Airflow? Your errors are also completely inconsistent. Your import error shows "/opt/airflow" which indicates that your airflow runs on Linux (in Docker Compose??) but you seem to b ve installng provider and run your check on Windows - those are two completely different environments. If you wan to run Docker Compose Airlfow, then you should extend your Docker Image https://airflow.apache.org/docs/docker-stack/build.html rather than install stuff localy.

I am guessing because you have not provided enough context.

You need to learn the difference and learn how Docker Docker Compose works if you want to mange Airflow this way.

**Reply by irfanajr17** [reply]
i have installed in docker with docker exec -it '''airflow name' bash
but still the same error appears
![image](https://user-images.githubusercontent.com/109498315/180966904-c342e14b-9e04-46d1-9774-26e35e7955ec.png)


do you have a sample DAG for microsoft sql server to google cloud storage?

**Reply by potiuk** [reply]
If we have, then you will find "example dags" in the docuementation. Just browse the docs and find mssql provider. It's generally a good idea to check documentation first rather than expect attention of people.
